kongswap_adaptor/tests/allowance_expires.rs

```diff
@@ -0,0 +1,420 @@
+        .expect("KONGSWAP_ADAPTOR_CANISTER_WASM_PATH must be set.");
+    Wasm::from_file(wasm_path)
+}
+
+async fn install_kong_adaptor(
+    pocket_ic: &PocketIc,
+    wasm: Wasm,
+    canister_id: Principal,
+    treasury_icp_account: sns_treasury_manager::Account,
+    treasury_sns_account: sns_treasury_manager::Account,
+    amount_icp_e8s: u64,
+    amount_sns_e8s: u64,
+) {
+    let sns_asset = Asset::Token {
+        symbol: "SNS".to_string(),
+        ledger_canister_id: *SNS_LEDGER_CANISTER_ID,
+        ledger_fee_decimals: Nat::from(FEE),
+    };
+
+    let icp_asset = Asset::Token {
+        symbol: "ICP".to_string(),
+        ledger_canister_id: *ICP_LEDGER_CANISTER_ID,
+        ledger_fee_decimals: Nat::from(FEE),
+    };
+
+    let arg = TreasuryManagerArg::Init(TreasuryManagerInit {
+        allowances: vec![
+            Allowance {
+                asset: sns_asset,
+                amount_decimals: Nat::from(amount_sns_e8s),
+                owner_account: treasury_sns_account,
+            },
+            Allowance {
+                amount_decimals: Nat::from(amount_icp_e8s),
+                asset: icp_asset,
+                owner_account: treasury_icp_account,
+            },
+        ],
+    });
+
+    let arg = candid::encode_one(&arg).unwrap();
+
+    pocket_ic
+        .install_canister(
+            canister_id,
+            wasm.bytes(),
+            arg,
+            Some(*SNS_GOVERNANCE_CANISTER_ID),
+        )
+        .await;
+
+    let subnet_id = pocket_ic.get_subnet(canister_id).await.unwrap();
+    println!(
+        "Installed the KongSwapAdaptor canister ({}) onto subnet {}",
+        canister_id, subnet_id
+    );
+}
+
+async fn install_kong_swap(pocket_ic: &PocketIc) {
+    // Install KongSwap
+    let wasm_path = std::env::var("KONG_BACKEND_CANISTER_WASM_PATH")
+        .expect("KONG_BACKEND_CANISTER_WASM_PATH must be set.");
+
+    let kong_backend_wasm = Wasm::from_file(wasm_path);
+
+    let controllers = vec![Principal::anonymous()];
+
+    let canister_id = *KONGSWAP_BACKEND_CANISTER_ID;
+
+    install_canister_with_controllers(
+        pocket_ic,
+        "KongSwap Backend Canister",
+        CanisterInstallationTarget::CanisterId(canister_id),
+        vec![],
+        kong_backend_wasm,
+        controllers,
+    )
+    .await;
+}
+
+async fn mint_tokens<Agent>(
+    agent: &mut Agent,
+    icrc1_ledger_canister_id: Principal,
+    beneficiary_account: Account,
+    amount_e8s: u64,
+) where
+    Agent: AbstractAgent,
+{
+    let request = TransferArg {
+        from_subaccount: None,
+        to: beneficiary_account,
+        fee: Some(Nat::from(0_u8)),
+        created_at_time: None,
+        memo: None,
+        amount: Nat::from(amount_e8s),
+    };
+
+    let _response = agent
+        .call(icrc1_ledger_canister_id, request)
+        .await
+        .unwrap()
+        .unwrap();
+}
+
+async fn install_sns_ledger(pocket_ic: &PocketIc) -> Principal {
```

Comment:
minor thing - I imagine this would be available already in another test - are there some common test utilities already?